arctic/store/bson_store.py

```diff
@@ -29,17 +29,23 @@ def __init__(self, arctic_lib):
         self._arctic_lib = arctic_lib
         self._reset()
 
-    @classmethod
-    def initialize_library(cls, arctic_lib, hashed=True, **kwargs):
+    def enable_sharding(self):
         logger.info("Trying to enable sharding...")
+        arctic_lib = self._arctic_lib
         try:
-            if not hashed:
-                logger.warning("Ignored hashed=False when enabling sharding, only hashed=True "
-                               " makes sense when they key is an ObjectId")
             enable_sharding(arctic_lib.arctic, arctic_lib.get_name(), hashed=True, key='_id')
         except OperationFailure as exception:
-            logger.warning(("Library created, but couldn't enable sharding: "
-                            "%s. This is OK if you're not 'admin'"), exception)
+            logger.warning("Could not enable sharding: %s, you probably need admin permissions.", exception)
+
+    @classmethod
+    def initialize_library(cls, arctic_lib, hashed=True, **kwargs):
```

Comment:
nit: should you keep the warning about the useless hashed flag?

Reply:
I would remove it. It doesn't add any value, and you would never use ranged sharding for ObjectIDs.

Reply:
Removing it would kinda change the signature even though kwargs should swallow hashed=True up this even if provided after removing it from the function signature ... wdyt?

Reply:
Removed the hashed=True argument.